### PR TITLE
chore: upgrade nodejs version to 18.x

### LINKED ISSUE #8386: migrate to node 18
Node 16 is reaching EOL in September 2023. 
cc: @BrunoQuaresma

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Close https://github.com/coder/coder/issues/8386

### CODE REVIEW

File: .github/actions/setup-node/action.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ runs:
     - name: Setup Node
       uses: buildjet/setup-node@v3
       with:
-        node-version: 16.20.1
+        node-version: 18.17.0
```
Comment (matifali) [inline .github/actions/setup-node/action.yaml:16]:
```suggestion
        node-version: 18.x
```
As we are installing the latest 18.x release in Dockerfile should we also do this in ci?
@kylecarbs for a 2nd thought.

File: dogfood/Dockerfile [hunk 2]
```diff
@@ -190,6 +190,7 @@ RUN LAZYGIT_VERSION=$(curl -s "https://api.github.com/repos/jesseduffield/lazygi
 # Install frontend utilities
 RUN apt-get update && \
 	# Node.js (from nodesource) and Yarn (from yarnpkg)
+	curl -fsSL https://deb.nodesource.com/setup_18.x | sudo -E bash - &&\
```
Comment (matifali) [inline dogfood/Dockerfile:193]: Here we are installing the latest 18.x release.

**Review by matifali — COMMENTED** [review]
Other than this it's good to go.

**Review by kylecarbs — APPROVED** [review]

File: .github/actions/setup-node/action.yaml (hunk 1 above)
Comment (BrunoQuaresma) [reply .github/actions/setup-node/action.yaml:16]: Why not? The docker file is for the workspace and the CI should use the same version as well.

**Pull request merged**